primitives/src/witness.rs

```diff
@@ -588,6 +594,7 @@ mod test {
     fn single_empty_element() -> Witness { Witness::from([[0u8; 0]]) }
 
     #[test]
+    #[cfg(feature = "std")]
     fn witness_debug_can_display_empty_element() {
```

Comment:
This whole test is a bit bogus using `println`. Perhaps you could use this

```rust
    #[test]
    #[cfg(feature = "alloc")]
    fn witness_debug_can_display_empty_element() {
        let witness = single_empty_element();

        let want = "Witness { num_elements: 1, total_bytes: 0, elements: [] }";
        let got = format!("{:?}", witness);
        assert_eq!(got, want);
    }
```

Reply:
This change could be added as a separate patch up front.

Reply:
We could do `let _ = format!("{:?}", witness)`.  But I think the existing test is probably fine. We may want to change it if we ever turn on the "no printfs in tests" clippy lint.

There are a bunch of types where our debug (and display) output changes depending whether you have an allocator.